File: src/main/java/com/vaadin/demo/component/emailfield/EmailFieldConstraints.java
package com.vaadin.demo.component.emailfield;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.EmailField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("email-field-constraints")
public class EmailFieldConstraints extends HorizontalLayout {

    public EmailFieldConstraints() {
        setPadding(false);

        // tag::snippet[]
        EmailField field = new EmailField("Email address");
        field.setRequiredIndicatorVisible(true);
        field.setPattern("^.+@example\\.com$");
        // end::snippet[]
        field.setHelperText("Only example.com addresses allowed");

        add(field);
    }

    public static class Exporter extends DemoExporter<EmailFieldConstraints> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/emailfield/email-field-constraints.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/email-field';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('email-field-constraints')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-email-field
        pattern="^.+@example\\.com$"
        required
        min-length="13"
        max-length="20"
        label="Email address"
        error-message="Enter a valid example.com email address"
        helper-text="Only example.com addresses allowed"
      ></vaadin-email-field>
      <!-- end::snippet[] -->
    `;
  }
}
